pkg/stringutils/stringutils.go

```diff
@@ -33,14 +36,52 @@ func GenerateRandomASCIIString(n int) string {
 // Ellipsis truncates a string to fit within maxlen, and appends ellipsis (...).
 // For maxlen of 3 and lower, no ellipsis is appended.
 func Ellipsis(s string, maxlen int) string {
-	r := []rune(s)
-	if len(r) <= maxlen {
+	byteLen := len(s)
+	if byteLen == utf8.RuneCountInString(s) {
+		r := []rune(s)
+		if byteLen <= maxlen {
+			return s
+		}
+		if maxlen <= 3 {
+			return string(r[:maxlen])
+		}
+		return string(r[:maxlen-3]) + "..."
+	}
+
+	var display []int
+	displayLen := 0
+	rs := []rune(s)
+	for _, r := range rs {
+		//In a broad sense, wide characters include East Asian Wide, East Asian Fullwidth, and East Asian Ambiguous, 
+		//see http://unicode.org/reports/tr11/
+		EastAsianAmbiguous := 1
```

Comment:
Therese are already defined as `width.EastAsianAmbiguous`

Reply:
Thx for your help, I correct this, but the vendor problem still exists. @_@